engine/core/src/main/java/org/datacleaner/descriptors/RemoteDescriptorProvider.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.descriptors;
+
+import org.datacleaner.job.concurrent.TaskRunner;
+import org.datacleaner.job.tasks.Task;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * @Since 9/8/15
+ */
+public class RemoteDescriptorProvider extends AbstractDescriptorProvider {
+
+    private String url, username, password;
+    private CountDownLatch latch;
+
+    private final Map<String, AnalyzerDescriptor<?>> _analyzerBeanDescriptors = new HashMap<String, AnalyzerDescriptor<?>>();
+    private final Map<String, FilterDescriptor<?, ?>> _filterBeanDescriptors = new HashMap<String, FilterDescriptor<?, ?>>();
+    private final Map<String, TransformerDescriptor<?>> _transformerBeanDescriptors = new HashMap<String, TransformerDescriptor<?>>();
+    private final Map<String, RendererBeanDescriptor<?>> _rendererBeanDescriptors = new HashMap<String, RendererBeanDescriptor<?>>();
+
+    public RemoteDescriptorProvider(TaskRunner taskRunner, String url, String username, String password) {
+        this.url = url;
+        this.username = username;
+        this.password = password;
+        latch = new CountDownLatch(1);
+        taskRunner.run(new Task() {
```

Comment:
Instead of using TaskRunner here I would suggest using a LazyRef. Normally we only use the task runner at the job-execution level and in this case I guess you _always_ want multithreading (whereas TaskRunner allows e.g. singlethreaded implementations). The LazyRef also uses a shared thread pool and it has methods built-in for checking the status, so you can also get rid of the CountDownLatch in here.